oci/layout/oci_transport.go

```diff
@@ -186,9 +186,6 @@ func (ref ociReference) getManifestDescriptor() (imgspecv1.Descriptor, error) {
 	} else {
 		// if image specified, look through all manifests for a match
 		for _, md := range index.Manifests {
```

Comment:
I don't think I understand what you mean here.  If you're asking if we need to be able to read an image index, when it's the only manifest, by specifying an empty image name, then I think so, since we already supported reading single-image manifests that way.

Reply:
Yes, we support reading single-image manifests with no name annotation when it’s the only manifest in the repo.

---

I was thinking about cases like this:

Top-level index:
- unnamed descriptor for x86_64
- unnamed descriptor for arm64

(i.e. the top-level index is “the manifest list”; with no nested index entry in the top-level index)

or possibly

Top-level index:
- unnamed descriptor for a manifest list
  - unnamed descriptor for x86_64
  - unnamed descriptor for arm64
- unnamed descriptor for x86_64
- unnamed descriptor for arm64

(i.e. the output generated by `ociImageDestination` now, except with no name annotation).

---

Both cases would currently fail the `if len(index.Manifests) == 1 {` check.

Maybe it’s just not worth worrying about now.